### DISCUSSION TITLE
GPU support

### DISCUSSION THREAD

**Discussion by lavnishhh** [Zulko/moviepy#1949] [General]
Is it technically possible for moviepy to have gpu support?

**Comment by mgaitan** [comment]
yes, it should be as far `ffmpeg` has a builtin support for different gpus https://trac.ffmpeg.org/wiki/HWAccelIntro . 

we should try to do some experiments and benchmarks.

**Comment by kaushal101** [comment]
I have been trying to get this working, without much luck so far. I have opened up a question on stack overflow on this topic with some more details [here](https://stackoverflow.com/questions/77639677/moviepy-write-videofile-using-gpu-for-faster-encoding)

I'm creating a video from scratch using moviePY. I am generating all the required frames, adding required audio (including a voiceover and background music), adding a logo and finally writing the video file in 4K.

Everything works fine, except the write_videofile takes a lot of time.

I have read many related posts which mentions using the right codec etc. I have NVidia card, so tried both h264_nvenc and hevc_nvenc. Quality of the output video dropped with the first one, so I'm sticking to hevc_nvenc. I'm using below line to write the file.

video_clip.write_videofile(targetfile, codec="hevc_nvenc", threads=32, fps=24)

What I have noticed is, it does seem to be using the gpu, but very little. Compared to this, when I run stable diffusion or vegas rendering, it uses gpu a lot more.

That's why I think there is definitely some scope of improvement here. If you see below screenshot, when the video file write starts, the gpu utilisation increases a tiny bit, but it can take a lot more I think, isn't it?

I can try various parameters that I've seen in other threads, like logger=None, progress_bar = False, ffmpeg_params=['-b:v','10000k'] etc., but they are not going to improve gpu utilisation in any shape or form. I've been wondering what am I missing.

![tiny bumps noticed for GPU utilisation Screenshot 2023-12-11 113642](https://github.com/Zulko/moviepy/assets/12201711/97055fba-5bd6-4e4a-bb53-6cd73644e415)

**Reply by samuelbraun04** [reply]
Did you ever figure this out? I can't find any clear instructions on how to actually do this for moviepy.

**Reply by hetari** [reply]
I am unsure if using moviepy in this way is valid or helpful, but I suggest utilizing moviepy.video.io.ffmpeg_writer. and try to use it
